h5py/_hl/dataset.py

```diff
@@ -130,6 +130,11 @@ def make_new_dset(parent, shape=None, dtype=None, data=None,
         dcpl.set_obj_track_times(track_times)
     elif track_times is not None:
         raise TypeError("track_times must be either True or False")
+    if track_order in (True, False):
+        dcpl_crt_order.set_attr_creation_order(
```

Comment:
Does `track_order` need to make it into this expression somehow?

Reply:
Ooups :-)